packages/zwave-js/src/lib/commandclass/UserCodeCC.ts

```diff
@@ -756,7 +756,9 @@ export class UserCodeCC extends CommandClass {
 		}
 
 		// Synchronize user codes and settings
-		await this.refreshValues();
+		if (!this.driver.options.skipUserCodeInterview) {
```

Comment:
When we end up not querying the codes, we still need to create default metadata for each possible user code. Applications like zwavejs2mqtt and HomeAssistant operate mostly on values and need the metadata to discover the user codes.

Currently, this happens here https://github.com/zwave-js/node-zwave-js/blob/f5a353baa13334ea0c20f3ef3368710b9415e0e0/packages/zwave-js/src/lib/commandclass/UserCodeCC.ts#L274-L284, which is called when decoding `[Extended]UserCodeReports`.

We don't know for sure if the user codes are strings or Buffers, but I think it is safe to default to `string` before knowing what the user codes are like.

Reply:
hm interesting. i see what the problem is. Let me take a crack at writing a function that sets the metadata for each code even if we haven't queried the actual value from the device. I'm guessing this function will need to be used in a couple of spots (`persistUserCode` and `interview`).